### PR TITLE
py/gc: Shouldn't include mpconfig.h and misc.h in gc.h

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
since gc.h don't reference any symbol from these header files

### CODE REVIEW

**Comment by dpgeorge** [comment]
This looks like a neat simplification.  What was the reason to close the PR?

**Comment by xiaoxiang781216** [comment]
> This looks like a neat simplification. What was the reason to close the PR?

The old intent is to fix the lvgl porting, but I find the better method. But, this patch is general enough, I reopen it for your review..

**Comment by xiaoxiang781216** [comment]
@dpgeorge Why no log output from macos build:
https://github.com/micropython/micropython/pull/6763/checks?check_run_id=1820541944

**Comment by dpgeorge** [comment]
> Why no log output from macos build:

I'm not sure, it's been broken for a while.  You can ignore that.

File: ports/esp8266/gccollect.c [hunk 1]
```diff
@@ -26,6 +26,7 @@
 
 #include <stdio.h>
 
+#include "py/mpconfig.h"
```
Comment (dpgeorge) [inline ports/esp8266/gccollect.c:29]: is this needed?

Comment (xiaoxiang781216) [reply ports/esp8266/gccollect.c:29]: Yes, otherwise compiler complain that mp_uint_t isn't defined

Comment (dpgeorge) [reply ports/esp8266/gccollect.c:29]: yes, true!

**Comment by dpgeorge** [comment]
Thank you.  This has been merged in 5fdf351178df9a18df624ae0f5947d8a5a6bce40

**Pull request closed without merging**